.github/workflows/main.yml

```diff
@@ -0,0 +1,209 @@
+# Continuous integration and pull request validation builds for the
+# master and maintenance branches.
+name: CI Build
+
+on:
+  push:
+    branches: [ master, maint/* ]
+  pull_request:
+    branches: [ master, maint/* ]
+
+env:
+  docker-registry: docker.pkg.github.com
+  docker-config-path: azure-pipelines/docker
+
+jobs:
+  # Build the docker container images that we will use for our Linux
+  # builds.  This will identify the last commit to the repository that
+  # updated the docker images, and try to download the image tagged with
+  # that sha.  If it does not exist, we'll do a docker build and push
+  # the image up to GitHub Packages for the actual CI/CD runs.  We tag
+  # with both the sha and "latest" so that the subsequent runs need not
+  # know the sha.
+  build_containers:
+    name: Create docker image
+    strategy:
+      matrix:
+        container:
+        - { name: xenial, base: 'ubuntu:xenial' }
+        - { name: bionic, base: 'ubuntu:bionic' }
+        - { name: docurium, base: 'ubuntu:trusty' }
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out repository
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - name: Calculate image name
+      run: |
+        echo "::set-env name=docker-container::${{ github.repository }}/${{ matrix.container.name }}"
+        echo "::set-env name=docker-sha::$(git log -1 --pretty=format:"%h" -- ${{ env.docker-config-path }})"
```

Comment:
Is this to catch the case where e.g. the image changes and uses a different version of Docker or its config?